### PR TITLE
dev-python/pydotplus: PYTHON_COMPAT and EAPI update

### PR DESCRIPTION
**Pull request by TheChymera**
Bug: https://bugs.gentoo.org/695996
Package-Manager: Portage-3.0.2, Repoman-2.3.23
Signed-off-by: Horea Christian <chr@chymera.eu>

### CODE REVIEW

File: dev-python/pydotplus/pydotplus-2.0.2.ebuild [hunk 1]
```diff
@@ -14,25 +14,22 @@ SRC_URI="mirror://pypi/p/${PN}/${P}.tar.gz"
...
-#		dev-python/sphinx[${PYTHON_USEDEP}]
-#		dev-python/sphinx_rtd_theme[${PYTHON_USEDEP}]
-#		dev-python/tox[${PYTHON_USEDEP}]
-#		)
+IUSE="test"
+RESTRICT="!test? ( test )"
+
+DEPEND="dev-python/setuptools[${PYTHON_USEDEP}]
+	test? (
+		dev-python/pytest[${PYTHON_USEDEP}]
```
[the remark this answers is not in the document]
Comment (TheChymera) [reply dev-python/pydotplus/pydotplus-2.0.2.ebuild]: like so? seems strange to me to use `distutils_enable_tests pytest` when right afterwards I need to define the test phase to be something different from the pytest standard. What “duplication” are you referring to?

File: dev-python/pydotplus/files/pydotplus-2.0.2-tests.patch [hunk 2]
```diff
@@ -0,0 +1,26 @@
+--- a/test/pydot_unittest.py
++++ b/test/pydot_unittest.py
+@@ -18,7 +18,7 @@ import sys
+
+ import pydotplus
+ import unittest
+-
++import pytest
```
Comment (mgorny) [inline dev-python/pydotplus/files/pydotplus-2.0.2-tests.patch:8]: Sorry I didn't notice this before. So you're actually adding dependency on pytest? Why aren't you just using unittest's functions?

File: dev-python/pydotplus/pydotplus-2.0.2.ebuild [hunk 3]
```diff
@@ -1,38 +1,32 @@
...
-DEPEND="dev-python/setuptools[${PYTHON_USEDEP}]"
-#	test? (
-#		dev-python/flake8[${PYTHON_USEDEP}]
-#		dev-python/pytest[${PYTHON_USEDEP}]
-#		dev-python/pytest-cov[${PYTHON_USEDEP}]
-#		dev-python/sphinx[${PYTHON_USEDEP}]
-#		dev-python/sphinx_rtd_theme[${PYTHON_USEDEP}]
-#		dev-python/tox[${PYTHON_USEDEP}]
-#		)
+IUSE="test"
```
Comment (mgorny) [inline dev-python/pydotplus/pydotplus-2.0.2.ebuild:17]: Redundant.

File: dev-python/pydotplus/pydotplus-2.0.2.ebuild [hunk 4]
```diff
@@ -1,38 +1,32 @@
...
-#the required files:
-#https://github.com/carlos-jenkins/pydotplus/issues/12
-#python_test() {
-#	${EPYTHON} -m unittest discover || die
-#	tox
-#}
+PATCHES=( "${FILESDIR}/${P}-tests.patch" )
+
+distutils_enable_tests pytest
+python_test(){
```
Comment (mgorny) [inline dev-python/pydotplus/pydotplus-2.0.2.ebuild:28]: Empty line above, space after `()`.

**Review by mgorny — CHANGES_REQUESTED** [review]

**Comment by TheChymera** [comment]
@mgorny ok

File: dev-python/pydotplus/pydotplus-2.0.2.ebuild [hunk 5]
```diff
@@ -1,38 +1,29 @@
...
-#	test? (
-#		dev-python/flake8[${PYTHON_USEDEP}]
-#		dev-python/pytest[${PYTHON_USEDEP}]
-#		dev-python/pytest-cov[${PYTHON_USEDEP}]
-#		dev-python/sphinx[${PYTHON_USEDEP}]
-#		dev-python/sphinx_rtd_theme[${PYTHON_USEDEP}]
-#		dev-python/tox[${PYTHON_USEDEP}]
-#		)
+
+DEPEND=""
```
Comment (mgorny) [inline dev-python/pydotplus/pydotplus-2.0.2.ebuild:18]: Remove the empty var.

File: dev-python/pydotplus/pydotplus-2.0.2.ebuild [hunk 6]
```diff
@@ -1,38 +1,29 @@
...
-#python_test() {
-#	${EPYTHON} -m unittest discover || die
-#	tox
-#}
+PATCHES=( "${FILESDIR}/${P}-tests.patch" )
+
+python_test() {
+	pushd test > /dev/null || die
+		python pydot_unittest.py || die
+	popd > /dev/null
```
Comment (mgorny) [inline dev-python/pydotplus/pydotplus-2.0.2.ebuild:28]: ||die

**Review by mgorny — COMMENTED** [review]
Are you *sure* this package doesn't need to depend on graphviz? Also, you are referencing the wrong bug.

File: dev-python/pydotplus/pydotplus-2.0.2.ebuild (hunk 5 above)
Comment (TheChymera) [reply dev-python/pydotplus/pydotplus-2.0.2.ebuild:18]: ok

File: dev-python/pydotplus/pydotplus-2.0.2.ebuild (hunk 6 above)
Comment (TheChymera) [reply dev-python/pydotplus/pydotplus-2.0.2.ebuild:28]: most ebuilds have it without die (232 without, 217 with):

```
chymera@silenthost ~/src/pydotplus $ qgrep "popd > /dev/null" | wc -l
449
chymera@silenthost ~/src/pydotplus $ qgrep "popd > /dev/null || die" | wc -l
217
```

Why is this required here? doesn't the dir just change to the correct one after the function exits anyway?

**Comment by TheChymera** [comment]
@mgorny ping?

**Comment by TheChymera** [comment]
@mgorny ping?

**Review by mgorny — APPROVED** [review]
Thanks.

**Pull request closed without merging**